src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,32 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    Animal.alive.push(this);
+  }
+
+  health = 100;
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hidden = false;
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(victim) {
+    if (victim.hidden !== true && victim.__proto__ !== Carnivore.prototype) {
```

Comment:
[CODE QUALITY] The check for `victim.hidden !== true` is correct, but checking `victim.__proto__ !== Carnivore.prototype` is not the preferred way to determine if the `victim` is an instance of `Carnivore`. It's recommended to use `instanceof` operator for this purpose, as it is clearer and more idiomatic in JavaScript.

Reply:
> [CODE QUALITY] The check for `victim.hidden !== true` is correct, but checking `victim.__proto__ !== Carnivore.prototype` is not the preferred way to determine if the `victim` is an instance of `Carnivore`. It's recommended to use `instanceof` operator for this purpose, as it is clearer and more idiomatic in JavaScript.

Good point
And you don't need to compare with true as the value is already boolean